lib/insights_cloud/async/insights_scheduled_sync.rb

```diff
@@ -13,9 +13,17 @@ def plan
           return
         end
 
-        after_delay do
-          plan_full_sync
+        # rubocop:disable Style/GuardClause
+        unless ForemanRhCloud.with_local_advisor_engine?
+          after_delay do
+            plan_full_sync
+          end
         end
+        # rubocop:enable Style/GuardClause
+      end
+
+      def run
+        output[:status] = _('The scheduled process is disabled because this Foreman is configured with the use_local_advisor_engine option.') if ForemanRhCloud.with_local_advisor_engine?
       end
```

Comment:
As far as I remember, unless you call `plan_self` in `plan`, the `run` will never be executed, so why have it here? Or am I misremembering?

Reply:
I'm pretty sure you're not misremembering ;)

@chris1984 we can verify this during testing, but I'm pretty sure `plan_self` will be needed in `plan`, from what I remember.